File: Search.SearchService/Internal/RequestComparer.cs
using System.Collections.Generic;

namespace Search.SearchService.Internal
{
    internal class RequestComparer : IEqualityComparer<SearchRequest>
    {
        public bool Equals(SearchRequest x, SearchRequest y)
        {
            return Equals(x.From, y.From)
                && Equals(x.Size, y.Size)
                && Equals(x.Query, y.Query);
        }

        public int GetHashCode(SearchRequest obj)
        {
            return unchecked(
                obj.Query.GetHashCode() * 2081561 +
                obj.From.GetHashCode() * 61583 +
                obj.Size.GetHashCode());
        }
    }
}

File: FSharpSearchService/RequestComparer.fs
namespace FSharpSearchService

open System.Collections.Generic

type RequestComparer()=
    interface IEqualityComparer<SearchRequest> with
        member this.Equals(x: SearchRequest, y: SearchRequest): bool = 
            (obj.Equals(x.form,y.form)&&obj.Equals(x.Size, y.Size)&&obj.Equals(x.Query, y.Query))
        member this.GetHashCode(obj: SearchRequest): int = 
            obj.query.GetHashCode()*2081561+obj.form.GetHashCode()* 61583+obj.size.GetHashCode()